### PR TITLE
[Qwen] Rewrite RmsNorm and Embedding layer for Qwen2/3

### PR DESCRIPTION
**Pull request by lk-chen**
# Description

Follow up #1484 , see [RFC](#1485 ). After this PR, the name `mapping` in both files are empty, we can refactor weight loading logic afterwards (#1479 )

# Tests

1. CI

2. benchmark / eval
```
# serve
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
MODEL_IMPL_TYPE=flax_nnx vllm serve $CURRENT_VLLM_MODEL \
    --max-model-len=8192 \
    --disable-log-requests \
    --no-enable-prefix-caching

# benchmark
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
vllm bench serve \
    --model $CURRENT_VLLM_MODEL \
    --num-prompts 1000 \
    --dataset-name random \
    --random-input-len 1024 \
    --random-output-len 4096

# eval
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
lm_eval \
    --model local-completions \
    --model_args base_url=http://localhost:8000/v1/completions,model=$CURRENT_VLLM_MODEL \
    --tasks gsm8k_cot \
    --batch_size auto \
    --apply_chat_template \
    --num_fewshot 8
```

before: see https://github.com/vllm-project/tpu-inference/pull/1484
after:
```
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Benchmark duration (s):                  1683.56   
Total input tokens:                      1024000   
Total generated tokens:                  4096000   
Request throughput (req/s):              0.59      
Output token throughput (tok/s):         2432.94   
Peak output token throughput (tok/s):    3584.00   
Peak concurrent requests:                1000.00   
Total token throughput (tok/s):          3041.18                <-------------- - 0.1%
---------------Time to First Token----------------
Mean TTFT (ms):                          617639.52 
Median TTFT (ms):                        630893.67 
P99 TTFT (ms):                           1348286.18
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          86.89     
Median TPOT (ms):                        75.35     
P99 TPOT (ms):                           127.25    
---------------Inter-token Latency----------------
Mean ITL (ms):                           86.89     
Median ITL (ms):                         64.90     
P99 ITL (ms):                            124.05    
==================================================
```
|  Tasks  |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot|      3|flexible-extract|     8|exact_match|↑  |0.7582|±  |0.0118|
|         |       |strict-match    |     8|exact_match|↑  |0.0629|±  |0.0067|

3. 2.5-vl model can serve
```
export CURRENT_VLLM_MODEL=Qwen/Qwen2.5-VL-3B-Instruct
MODEL_IMPL_TYPE=flax_nnx vllm serve $CURRENT_VLLM_MODEL \
    --max-model-len=8192 \
    --disable-log-requests \
    --no-enable-prefix-caching \
    --max-num-batched-tokens 16384
```

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/layers/jax/embed.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        if name == "embedding":
+            # nnx.Embed needs to access self.embedding
+            return self.weight
+
+    def __call__(self, x) -> jax.Array:
+        if self.quant_method is None:
+            return super().__call__(x)
+        return self.quant_method.apply_jax(self, x)
+
+    def decode(self, x: jax.Array) -> jax.Array:
```
Comment (jrplatin) [inline tpu_inference/layers/jax/embed.py:57]: I assume this is needed because `weight` is `embedding` in the parent class' (`nnx.Embedding`) right? Or is there another reason?

File: tpu_inference/models/jax/qwen2.py [hunk 2]
```diff
@@ -301,7 +305,7 @@ def __call__(
         if inputs_embeds is not None:
             x = inputs_embeds
         else:
-            x = self.embed(input_ids)
+            x = self.embed_tokens(input_ids)
```
Comment (jrplatin) [inline tpu_inference/models/jax/qwen2.py:308]: Is there a reason we want to rename `embed` to `embed_tokens`?

**Review by jrplatin — APPROVED** [review]

File: tpu_inference/models/jax/qwen2.py (hunk 2 above)
Comment (lk-chen) [reply tpu_inference/models/jax/qwen2.py:308]: This is to be aligned with HF param naming, and eventually get rid of name mapping below L375
image

File: tpu_inference/layers/jax/embed.py (hunk 1 above)
Comment (lk-chen) [reply tpu_inference/layers/jax/embed.py:57]: This is needed besides `__call__` because JaxEmbed is used twice in a model:
- once at the beginning to turn id->embed, which is done in `__call__`
- another at the end to do projection to turn embed->logits then sampling afterwards, which is done through this function
